src/sidebar/test/oauth-auth-test.js

```diff
@@ -53,6 +53,18 @@ describe('oauth auth', function () {
       });
     });
 
+    it('should raise if a grant token request fails', function () {
```

Comment:
Sigh, I don't know! You post the grant token to the endpoint in order to get an access token back. Is this a grant token request or an access token request?

I suppose your are _requesting_ an access token as in that's the thing that you're asking for in response.

But that is also true of refresh token requests, so would you call those access token requests as well?

With the way I've done it, if you call the request after the token you're _sending_, then you have grant token requests and refresh token requests (no ambiguity).

I'm not sure how the spec names these though...